src/pages/index.js

```diff
@@ -0,0 +1,372 @@
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(cardSubmitBtn, false, "Save"); 
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(avatarSubmitBtn, true, "Save", "Saving...");
+
+  api.editAvatarInfo(avatarInput.value)  
+    .then((data) => {
+      avatarImage.src = data.avatar;
+      avatarForm.reset();
+      closeModal(avatarModal);
+
+      api.getAppInfo()
+        .then(([cards, userInfo]) => {
+          avatarImage.src = userInfo.avatar;  
+        })
+        .catch(console.error);  
+    })
+    .catch(console.error) 
+    .finally(() => {
+      setButtonText(avatarSubmitBtn, false, "Save");  
+    });
+}
+
+function closeOnOverlayClick(modal) {
+  modal.addEventListener("click", (evt) => {
+    if (evt.target === modal) {
+      closeModal(modal);
+    }
+  });
+}
+modals.forEach((modal) => {
+  closeOnOverlayClick(modal);
+});
+
+function handleEscape(evt) {
+  if (evt.key === "Escape") {
+    modals.forEach(closeModal);
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscape);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscape);
+}
+
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  resetValidation(
+    editModal,
+    [editModalNameInput],
+    settings
+  );
+  openModal(editModal);
+});
+
+editModalCloseBtn.addEventListener("click", () => {
+  closeModal(editModal);
+});
+
+cardModalBtn.addEventListener("click", () => {
+  cardForm.reset();
+  cardSubmitBtn.disabled = true; 
+  cardSubmitBtn.classList.add("modal__submit-btn_disabled"); 
+  openModal(cardModal);
+});
+
+cardModalCloseBtn.addEventListener("click", () => {
+  closeModal(cardModal);
+});
+
+editFormElement.addEventListener("submit", handleEditFormSubmit);
+cardForm.addEventListener("submit", handleAddCardSubmit);
+
+avatarModalBtn.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
+avatarModalCloseBtn.addEventListener("click", () => {
+  closeModal(avatarModal);
+});
+
+avatarForm.addEventListener("submit", handleAvatarSubmit);
+deleteForm.addEventListener("submit", handleDeleteSubmit);
+
+document.querySelectorAll(".modal__close-btn").forEach((button) => {
+  button.addEventListener("click", (event) => {
+    const modal = event.target.closest(".modal");
+    closeModal(modal);
+  });
+});
+
+document.querySelectorAll(".modal__submit-btn_cancel").forEach((button) => {
+  button.addEventListener("click", (event) => {
+    const modal = event.target.closest(".modal");
+    closeModal(modal);
+  });
+});
+
+document.querySelectorAll(".modal").forEach((modal) => {
+  modal.addEventListener("click", (event) => {
+    if (event.target.classList.contains("modal")) {
+      closeModal(modal);
+    }
+  });
+});
+
+document.addEventListener("keydown", (event) => {
```

Comment:
Please, delete it

You have the `handleEscape` function for that. And you do it in `openModal` correctly